libs/orchestrator/json_models.go

```diff
@@ -12,19 +16,21 @@ type StageJson struct {
 }
 
 type JobJson struct {
-	ProjectName       string            `json:"projectName"`
-	ProjectDir        string            `json:"projectDir"`
-	ProjectWorkspace  string            `json:"projectWorkspace"`
-	Terragrunt        bool              `json:"terragrunt"`
-	Commands          []string          `json:"commands"`
-	ApplyStage        StageJson         `json:"applyStage"`
-	PlanStage         StageJson         `json:"planStage"`
-	PullRequestNumber *int              `json:"pullRequestNumber"`
-	EventName         string            `json:"eventName"`
-	RequestedBy       string            `json:"requestedBy"`
-	Namespace         string            `json:"namespace"`
-	StateEnvVars      map[string]string `json:"stateEnvVars"`
-	CommandEnvVars    map[string]string `json:"commandEnvVars"`
+	ProjectName        string                            `json:"projectName"`
+	ProjectDir         string                            `json:"projectDir"`
+	ProjectWorkspace   string                            `json:"projectWorkspace"`
+	Terragrunt         bool                              `json:"terragrunt"`
+	Commands           []string                          `json:"commands"`
+	ApplyStage         StageJson                         `json:"applyStage"`
+	PlanStage          StageJson                         `json:"planStage"`
+	PullRequestNumber  *int                              `json:"pullRequestNumber"`
+	EventName          string                            `json:"eventName"`
+	RequestedBy        string                            `json:"requestedBy"`
+	Namespace          string                            `json:"namespace"`
+	StateEnvVars       map[string]string                 `json:"stateEnvVars"`
+	CommandEnvVars     map[string]string                 `json:"commandEnvVars"`
+	StateEnvProvider   *stscreds.WebIdentityRoleProvider `json:"stateEnvProvider"`
```

Comment:
This is serialization at the point of orchestrator, we won't be able to serialize providers back to the cli, so we should only store the roles as strings

StateEnvRole string
CommandEnvRole string